.github/workflows/_mac-test-mps.yml

```diff
@@ -32,17 +32,21 @@ jobs:
           git clean -fxd
 
       - name: Download build artifacts
+        id: download-build-artifacts
         uses: ./.github/actions/download-build-artifacts
         with:
           name: ${{ inputs.build-environment }}
           use-gha: true
+        continue-on-error: true
```

Comment:
Why do we want to continue on error here?

Reply:
This is an edge case in which only mps tests are run in this workflow file.  It doesn't follow the same convention as other configs in the test matrix.  Specifically, if you look at `_mac-test.yml`, it has a test-matrix input while this one doesn't.

When you select `functorch` tests, for example, it makes sense not to run mps tests.  The step here tries to download the build artifacts, which don't exist in this case because the build step isn't needed at all for `functorch`.   So, it's ok to continue and not fail the workflow.

Reply:
As discussed offline, let's move this logic to the trunk + build steps so we reliably know when we should skip this entire workflow